python/hyperspace/hyperspace.py

```diff
@@ -0,0 +1,172 @@
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.deleteIndex("indexname")
+        """
+        self.hyperspace.deleteIndex(indexName)
+
+    def restoreIndex(self, indexName):
+        """
+        Restores index with given index name.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.restoreIndex("indexname")
+        """
+        self.hyperspace.restoreIndex(indexName)
+
+    def vacuumIndex(self, indexName):
+        """
+        Vacuums index with given index name.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.vacuumIndex("indexname")
+        """
+        self.hyperspace.vacuumIndex(indexName)
+
+    def refreshIndex(self, indexName):
+        """
+        Update indexes for the latest version of the data.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.refreshIndex("indexname")
+        """
+        self.hyperspace.refreshIndex(indexName)
+
+    def cancel(self, indexName):
+        """
+        Cancel api to bring back index from an inconsistent state to the last known stable state.
+        :param indexName: index name
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> hyperspace.cancel("indexname")
+        """
+        self.hyperspace.cancel(indexName)
+
+    def explain(self, df, redirectFunc=lambda x: sys.stdout.write(x), verbose=False):
+        """
+        Explains how indexes will be applied to the given dataframe.
+        :param df: dataFrame
+        :param redirectFunc: optional function to redirect output of explain
+
+        >>> hyperspace = Hyperspace(spark)
+        >>> df = spark.read.parquet("./sample.parquet").toDF("c1", "c2", "c3")
+        >>> hyperspace.explain(df)
+        """
+        analyzer = self.jvm.com.microsoft.hyperspace.index.plananalysis.PlanAnalyzer
+        result_string = analyzer.explainString(df._jdf, self.spark._jsparkSession,
+                                               self.hyperspace.indexes(), verbose)
+        redirectFunc(self.jvm.java.lang.String(result_string))
+
+    @staticmethod
+    def enable(spark):
+        """
+        Enables hyperspace index usage on given spark session.
+        :param spark: sparkSession
+
+        >>> Hyperspace.enable(spark)
+        """
+        spark._jvm.com.microsoft.hyperspace.util.PythonUtils.enableHyperspace(spark._jsparkSession)
+        return spark
+
+    @staticmethod
+    def disable(spark):
+        """
+        Disables hyperspace index usage on given spark session.
+        :param spark: sparkSession
+
+        >>> Hyperspace.disable(spark)
+        """
+        spark._jvm.com.microsoft.hyperspace.util.PythonUtils.disableHyperspace(spark._jsparkSession)
+        return spark
+
+    @staticmethod
+    def isEnabled(spark):
+        """
+        Checks if hyperspace is enabled or not.
```

Comment:
nit: Can you make the usage of the word hyperspace consistent throughout? It should be `Hyperspace`

Reply:
Updated thanks!